src/App.tsx

```diff
@@ -26,70 +28,116 @@ type ReorderOptions = {
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  reverseGoods = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortGoodsByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  sortGoodsAlpha = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classnames({
+              button: true,
+              'is-info': true,
+              'is-light': this.state.sortType !== SortType.ALPABET,
+            })}
+            onClick={this.sortGoodsAlpha}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={this.sortGoodsByLength}
+            className={classnames({
+              button: true,
+              'is-success': true,
+              'is-light': this.state.sortType !== SortType.LENGTH,
+            })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classnames({
+              button: true,
+              'is-warning': true,
+              'is-light': !this.state.isReversed,
+            })}
+            onClick={this.reverseGoods}
+          >
+            Reverse
+          </button>
+
+          {(this.state.sortType !== SortType.NONE || this.state.isReversed) && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={this.reset}
+            >
+              Reset
+            </button>
+          )}
+        </div>
+        <List goods={getReorderedGoods(goodsFromServer, this.state)} />
```

Comment:
Where is the file with this component? Did you forget to add it?

Reply:
strange, yep, my bad, sor

Reply:
Link component is still absent
![image](https://user-images.githubusercontent.com/96409563/201890560-f091beb9-ce50-4148-82a6-e38db5b8972e.png)